firedrake/assemble.py

```diff
@@ -393,7 +393,7 @@ def thunk(bcs):
             loops.append(zero_tensor)
         else:
             zero_tensor()
-        for indices, (kernel, integral_type, needs_orientations, subdomain_id, domain_number, coeff_map, needs_cell_facets, pass_layer_arg) in kernels:
+        for indices, (kernel, integral_type, needs_orientations, subdomain_id, domain_number, coeff_map, needs_cell_facets, pass_layer_arg, needs_cell_sizes) in kernels:
```

Comment:
This is now really begging for something more descriptive than an n-tuple... (`namedtuple`?)

Reply:
So in tsfc_interface it is a named tuple, I should destructure by name here.